### ISSUE TITLE
Question: How to create a "service only" container 

### ISSUE THREAD

**Issue by KristofKlein** [microsoft/navcontainerhelper#3212]
Not a bug but a question:

assuming I want to have BC Container that only lives for working on Job Queue Tasks, or NAS. So I don't want IIS, nor Web Services. Not even a SQL Express. Just a container, I hook up with a DB and that's it it runs until I recreate it or shut down. 

What's the quickest way to do so? I recall all those "hooks" or "script overwrites". Guess the answer is somewhere in that? 
Or would be a "make your own docker image" a better way? 

Some help would be appreciated :)

**Comment by jonaswre** [comment]
To be honest I think starting from Scratch is the best option. It gives you the most control over where what should be installed when. But you could borrow from bccontainerhelper some of the scripts.

The containers are not meant to be used for such a case they were designed do be easy to deploy dev environments and in that break some of the stuff that makes docker great.

**Comment by tfenster** [comment]
@KristofKlein If you define the connection to an external SQL database and set the env variable `webClient=N`, then you don't get (running) SQL and IIS (see https://github.com/microsoft/nav-docker/blob/master/generic/Run/navstart.ps1#L97-L115). And with the env setting `customNavSettings=SOAPServicesEnabled=false,ODataServicesEnabled=false` you can disable SOAP and REST. Sure, all the files and functionality are in the image (can't remember, maybe you can even tweak the build process here), but during runtime you would already have only the NST with a few config settings

**Comment by KristofKlein** [comment]
see - it is all there :D Thanks @tfenster ! I will give this a try. Guess this will be just what I was looking for. If spinning up takes time - pfff - It's a windows container - it always takes time :D

**Comment by jonaswre** [comment]
@tfenster out of curiosity. They are still installed, just not started right?

**Comment by tfenster** [comment]
@jonaswre Yes, 100 % correct. Those settings only configure whether the services are started or not. But if you want to go the full way and not even have them installed, you could add an additional build argument and only conditionally run https://github.com/microsoft/nav-docker/blob/master/generic/DOCKERFILE#L14-L31. If those are skipped, you should have a "pure BC" image without SQL and IIS. Maybe some of https://github.com/microsoft/nav-docker/blob/master/generic/DOCKERFILE#L14 is needed for BC itself, but that should be easy enough to figure. Actually, an interesting experiment, might give that a try myself :)

**Comment by KristofKlein** [comment]
Do it :) If you need a tester let me know :D

**Comment by tfenster** [comment]
@KristofKlein I couldn't resist... Can you please try to

- clone https://github.com/tfenster/nav-docker/ --> this has my changes to allow creating an image with only BC
- go to folder "generic"
- create a file "settings.ps1" with content `$bcOnly = "true"` --> make sure the following command runs a build with only BC
- run build.ps1 --> this builds the generic image without SQL and IIS
- run `New-BcImage -artifactUrl (Get-BCArtifactUrl -type OnPrem -country de -version 23) -baseImage mygeneric:1.1.0.0 -imageName mybc` --> this builds the specific image for BC 23 DE based on the generic image that you just created
- run `docker run -e accept_eula=y -e accept_outdated=y -e databaseUserName=sa -e databasePassword=Super5ecret! -e databaseServer=172.24.7.162 -e databaseInstance= -e databaseName=CronusDE -e webClient=N -e httpSite=N -e username=admin -e password=Super5ecret! -e usessl=n --name=bc mybc:onprem-23.0.12034.12841-de` --> this runs a container based on the specific image that you just created. Of course, you will have to change the database stuff (user, password, server, instance, name) to what is correct for you

If I do that, I seem to get a working BC instance (`Get-NAVServerInstance BC` returns fine), so I wonder what happens for you :)

**Comment by tfenster** [comment]
@KristofKlein a bit more detail here https://github.com/microsoft/navcontainerhelper/issues/3212 😊

**Comment by rdebath** [comment]
Just as a counterpoint to this I would estimate that a copy&paste install of the service tier is actually less intrusive than a similar docker install on the host.

So if you only want a tier for task-scheduler/odata or similar.
0) Make sure Dotnet 4.x is installed (default on svr 2022)
1) Copy the service tier directory to somewhere useful. 
2) Install a service for the default service tier with SC.
3) Setup the .config file and start it.
4) Enjoy.

You probably don't need the other prerequisites and for most of them you get an obvious error if you need it.
(The main one you might need is the report runner; if you're creating report PDFs in the background)
